src/main/java/org/openstreetmap/atlas/checks/distributed/ShardedIntegrityChecksSparkJob.java

```diff
@@ -0,0 +1,434 @@
+                    .collect(Collectors.toSet());
+            logger.error(
+                    "Unable to find standardized named shard files in the path {}/<countryName> for the countries {}. \n Files must be in format <country>_<zoom>_<x>_<y>.atlas",
+                    input, missingCountries);
+            return;
+        }
+
+        // List of spark RDDS/tasks
+        final List<JavaPairRDD<String, UniqueCheckFlagContainer>> countryRdds = new ArrayList<>();
+
+        // Countrify spark parallelization for better debugging
+        try (Pool checkPool = new Pool(countryShards.size(), "Countries Execution Pool"))
+        {
+            for (final Map.Entry<String, List<Shard>> countryShard : countryShards.entrySet())
+            {
+                checkPool.queue(() ->
+                {
+                    // Generate a task for each shard group
+                    final List<ShardedCheckFlagsTask> tasksForCountry = countryShard.getValue()
+                            .stream()
+                            .map(group -> new ShardedCheckFlagsTask(countryShard.getKey(), group,
+                                    this.countryChecks.get(countryShard.getKey())))
+                            .collect(Collectors.toList());
+
+                    // Set spark UI job title
+                    this.getContext().setJobGroup("0",
+                            String.format("Running checks on %s",
+                                    tasksForCountry.get(0).getCountry()));
+                    // Run each task in spark, producing UniqueCheckFlagContainers
+                    final JavaPairRDD<String, UniqueCheckFlagContainer> rdd = this.getContext()
+                            .parallelize(tasksForCountry, tasksForCountry.size())
+                            .mapToPair(produceFlags(input, output, this.configurationMap(),
+                                    fileHelper, shardingBroadcast, distanceToLoadShards,
+                                    (Boolean) commandMap.get(MULTI_ATLAS)));
+                    // Save the RDD
+                    rdd.persist(storageLevel);
+                    // Use a fast spark action to overcome spark lazy elocution. This is necessary
+                    // to get the UI title to appear in the right spot.
+                    rdd.count();
+                    // Add the RDDs to the list
+                    countryRdds.add(rdd);
+                });
+            }
+        }
+
+        // Set up for unioning
+        final JavaPairRDD<String, UniqueCheckFlagContainer> firstCountryRdd = countryRdds.get(0);
+        countryRdds.remove(0);
+
+        // Add UI title
+        this.getContext().setJobGroup("0", "Conflate flags and generate outputs");
+        // union the RDDs, combine the UniqueCheckFlagContainers by country, and proccess the flags
+        // through the output event service
+        this.getContext().union(firstCountryRdd, countryRdds)
```

Comment:
Is getting Checks per country not going to be supported moving forward? // Is it depended on the being able to access the persisted RDD?

Reply:
I'm not sure I exactly follow the question, but all this has to do with creating job labels for the Spark UI. In order to run and label each country as a different job in spark this is persisting the RDD, calling count (just to force execution), and then unioning the results to get everything into one stream. So checks are fully by country.